src/main/java/org/kohsuke/github/GraphQLDiscussion.java

```diff
@@ -0,0 +1,202 @@
+package org.kohsuke.github;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.kohsuke.github.internal.EnumUtils;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Date;
+import java.util.Objects;
+
+/**
+ * A discussion in the project.
+ * <p>
+ * This is different from Teams discussions (see {@link GHDiscussion}).
+ * <p>
+ * The discussion event exposes the GraphQL object directly. The new Discussions API is only available through GraphQL.
+ *
+ * @author Guillaume Smet
+ * @see <a href="https://docs.github.com/en/graphql/guides/using-the-graphql-api-for-discussions#discussion">The GraphQL
+ *      API for Discussions</a>
+ */
+public class GraphQLDiscussion {
```

Comment:
@bitwiseman I'm not sure about this class name. The specificity of it is that it's a GraphQL object injected in the REST API (the GitHub documentation here https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#discussion directly points to the GraphQL documentation). Also the Discussions API is not available as a REST API.

FTR, we already have a GHDiscussion object that is different as related to the Teams Discussions API and it's not the same.

I'm open to any suggestion as I'm not totally happy about this.